src/Voron/StorageEnvironment.cs

```diff
@@ -784,8 +784,12 @@ private void ThrowOnWriteTransactionOpenedByTheSameThread()
                 currentWriteTransactionHolder == NativeMemory.CurrentThreadStats)
             {
                 throw new InvalidOperationException($"A write transaction is already opened by thread name: " +
-                                                    $"{currentWriteTransactionHolder.Name}, Id: {currentWriteTransactionHolder.ManagedThreadId}");
+                                                    $"{currentWriteTransactionHolder.Name}, Id: {currentWriteTransactionHolder.ManagedThreadId}{Environment.NewLine}" +
+                                                    $"{currentWriteTransactionHolder.CapturedStackTrace}");
             }
+
+            if (currentWriteTransactionHolder != null)
+                currentWriteTransactionHolder.CapturedStackTrace = Environment.StackTrace;
```

Comment:
now I did that only for the server env, still heavy, but less :)

Reply:
Can we enable that only when running `RachisTests`? Can we enable that _only_ for specific test - `RavenDB_13659`?

Reply:
I believe it involve more than one test, but yes we can enable it only for `RachisTests`

Reply:
we don't reuse, but in the test we schedule `OpenWriteTransaction` in a timer callback, which uses a TP thread, which are shared across the process.

So I want to understand with which one we have a collision here. (it might be that the collision is in the test itself, but I was unable to repro this locally)

Reply:
We have discussed that we'll enable it only for a specific test